client/sites/components/sites-dashboard.tsx

```diff
@@ -193,44 +192,47 @@ const SitesDashboard = ( {
 				},
 			},
 		},
-	} as DataViewsState;
-	const [ dataViewsState, setDataViewsState ] = useState< DataViewsState >( defaultDataViewsState );
+	} as View;
+	const [ viewState, setViewState ] = useState< View >( defaultViewState );
 
 	useEffect( () => {
 		const fields = getFieldsByBreakpoint( isDesktop );
 		const fieldsForBreakpoint = [ ...fields ].sort().toString();
-		const existingFields = [ ...( dataViewsState?.fields ?? [] ) ].sort().toString();
+		const existingFields = [ ...( viewState?.fields ?? [] ) ].sort().toString();
 		// Compare the content of the arrays, not its referrences that will always be different.
 		// sort() sorts the array in place, so we need to clone them first.
 		if ( existingFields !== fieldsForBreakpoint ) {
-			setDataViewsState( ( prevState ) => ( { ...prevState, fields } ) );
+			setViewState( ( prevState ) => ( { ...prevState, fields } ) );
 		}
 
 		const siteNameColumnWidth = getSiteNameColWidth( isDesktop, isWide );
-		if ( dataViewsState.layout.styles.site.width !== siteNameColumnWidth ) {
-			setDataViewsState( ( prevState ) => ( {
+
+		if (
+			viewState.type === 'table' &&
+			viewState.layout?.styles?.site?.width !== siteNameColumnWidth
+		) {
+			// @ts-expect-error -- Need to fix the layout type upstream in @wordpress/dataviews to support styles.
+			setViewState( ( prevState ) => ( {
 				...prevState,
 				layout: {
 					styles: {
-						...prevState.layout.styles,
+						// @ts-expect-error -- Need to fix the layout type upstream in @wordpress/dataviews to support styles.
+						...prevState.layout?.styles,
 						site: {
 							width: siteNameColumnWidth,
 						},
 					},
 				},
 			} ) );
 		}
-	}, [ isDesktop, isWide, dataViewsState?.fields, dataViewsState?.layout?.styles?.site?.width ] );
-
-	useSyncSelectedSite( dataViewsState, setDataViewsState, selectedSite );
-
-	const { selectedSiteFeature, setSelectedSiteFeature } = useSyncSelectedSiteFeature( {
-		selectedSite,
-		initialSiteFeature,
-		dataViewsState,
-		featureToRouteMap: FEATURE_TO_ROUTE_MAP,
-		queryParamKeys: [ 'page', 'per-page', 'status', 'search', 'siteType' ],
```

Comment:
This removal will clear those query params when clicking on a site.

E.g. try this:

- In /sites, search for something
- Click on one of the results
- When the flyout panel appears, the expectation is that the `search` param should persist (along with `page` etc), so that the site list is showing the same results from before.

Reply:
Forgot about these. I think we can achieve the same by passing the current query params around.

The production implementation doesn't seem to fully cover the dataview options, should we try to cover these too?

These don't do anything in the collapsed view:
<img src="https://github.com/user-attachments/assets/1b29f589-3a41-4871-86d1-4db0b84c3e89" width=300 />

This ordering gets lost transitioning between views:
<img src="https://github.com/user-attachments/assets/cb8ba283-4512-47e8-acc2-84b1cce34beb" />

Reply:
> The production implementation doesn't seem to fully cover the dataview options, should we try to cover these too?

Interesting... I think it's because we used to NOT support those options (pre-DataViews era) and we just forgot to support those. 

> This ordering gets lost transitioning between views:

This does not seem to add anything to the query param, wondering if the current way will work? 🤔

Reply:
Might be new, one of the lines I saw was 2 months old. It sends the new order via onChangeView state so in theory should be able to store the order in a query param (even if it's just a csv of the headers in the latest order state).

Reply:
Query param passing solved by a1c366593118b753865c200eb0589b2bb0f7a518

Reply:
Added a function to wrap this, tried a couple different approaches didn't like any of them, this works for now but open to both new name suggestions and future iterations on this.

https://github.com/Automattic/wp-calypso/pull/96326/files?w=1#diff-666054b1e2a6f78aeb37afc6a89df6cd13092428043dfb620572824d2c6d15c3R85